lp2p/client/client_test.go

```diff
@@ -73,29 +72,30 @@ func TestGRPCClientTestFunc(t *testing.T) {
 
 	// test client
 	ctx, cancel := context.WithCancel(context.Background())
-	// for the initial 'get' to sync the chain
-	svc.(mock.MockService).EmitRand(false)
 	ch := c.Watch(ctx)
-	time.Sleep(100 * time.Millisecond)
 	for i := 0; i < 3; i++ {
+		// pub sub polls every 200ms
+		time.Sleep(250 * time.Millisecond)
 		svc.(mock.MockService).EmitRand(false)
-		fmt.Printf("round %d. emitting.\n", i)
+		fmt.Printf("round %d. emitted.\n", i)
 		select {
 		case r, ok := <-ch:
 			if !ok {
-				t.Fatal("expected randomness")
+				t.Fatal("expected randomness, watch outer channel was closed instead")
 			} else {
-				fmt.Print(r)
+				t.Log("received", r.Round())
 			}
-		case <-time.After(10 * time.Second):
+		// the period of the mock servers is 1 second
+		case <-time.After(5 * time.Second):
 			t.Fatal("timeout.")
 		}
 	}
 	svc.(mock.MockService).EmitRand(true)
 	cancel()
-	drain(t, ch, 10*time.Second)
+	drain(t, ch, 5*time.Second)
 }
 
+//nolint:unused
```

Comment:
if unused, why not kill it with 🔥 ?

If you right `click -> refactor -> change signature` in goland it will give you some magic to change it everywhere without having to do it by hand

Reply:
It's unused because I've set two test to `Skip` since they were flaky. But the goal is to patch the tests later.